pkg/lightning/config/config.go

```diff
@@ -527,6 +534,12 @@ func (cfg *Config) Adjust(ctx context.Context) error {
 		cfg.Mydumper.DefaultFileRules = true
 	}
 
+	// RegionConcurrency > NumCPU is meaningless.
+	cpuCount := runtime.NumCPU()
+	if cfg.App.RegionConcurrency > cpuCount {
```

Comment:
What about tidb backend? Maybe it useful for tidb backend since tidb io is the bottleneck..